### DISCUSSION TITLE
Operating in User's time zone

### DISCUSSION THREAD

**Discussion by vickkhera** [supabase/supabase#14084] [Questions]
One of the things that Postgres is exceptionally good at is time arithmetic and time zone conversions. When the data is stored as timestamp with timezone, it can be easily converted to any other time zone accurately within the DB, so queries can be relative to the user's time.

I am trying to find a way to get my app to operate in the local time zone of the user. To do so, I need to implement the equivalent of `set timezone = 'America/New_York';`.  The second runner up would be able to query a time using `at time zone 'America/New_York'`.

The most relevant discussion I found was on Github https://github.com/supabase/supabase/issues/176 from two years ago, but there's no solution for what I seek to do. I do not want to permanently set the timezone to anything other than UTC for the DB or the DB role. I do not want to do any data arithmetic in my app. I want to set it on a per-customer session (or query) basis because they could be anywhere in the country.

There's nothing obvious in the postgrest documentation or project repo how to do it, either. I would imagine an implementation doing it as part of the JWT or a special header to the REST request.

What are people doing to make the times in your app local to the user wherever they may be?

**Comment by shivavelingker** [comment]
I have the same question -- what did you end up doing? The hacky solution seems to be to store a separate timezone offset column, but maybe that's unnecessary?
